src/with_clause_parser.c

```diff
@@ -113,6 +114,22 @@ ts_with_clauses_parse(const List *def_elems, const WithClauseDefinition *args, S
 	return results;
 }
 
+extern TSDLLEXPORT char *
+ts_with_clause_result_deparse_value(const WithClauseResult *result)
+{
+	Oid oid = result->definition->type_id;
+	Assert(OidIsValid(oid));
+
+	Oid in_fn;
+	bool typIsVarlena;
+
+	getTypeOutputInfo(oid, &in_fn, &typIsVarlena);
+	Assert(OidIsValid(in_fn));
```

Comment:
This reads from PostgreSQL catalog tables, so a corrupt catalog table would then potentially cause a crash, which does not seem like a good idea. You would want to error out if the catalog tables are corrupt, however unlikely this is.
```suggestion
	getTypeOutputInfo(oid, &in_fn, &typIsVarlena);
	Ensure(OidIsValid(in_fn), "no output function for type with OID %d", oid);
```

Reply:
replaced it with `Assert`

I was looking around a bit to see what's the usual practice regarding this; and I've found some which doesn't really take care of this:
* [ts_internal_to_time_string](https://github.com/timescale/timescaledb/blob/f13214891cf47fed0db6183ce6b56d9273865de3/src/utils.c#L390-L392)
* [cagg_validate_query](https://github.com/timescale/timescaledb/blob/main/tsl/src/continuous_aggs/create.c#L1559-L1565)

and possibly there are a few more places

interesting that in Postgres's [ruleutils](https://github.com/postgres/postgres/blob/7da51590ed6cd46ff886e8e4d08e8703db9c2b5b/src/backend/utils/adt/ruleutils.c#L10405-L10408) this not checked during the deparse of const expressions; maybe they've omitted the check because the expr is most likely was just parsed and the fn for deparsing should also exists - but I was really expecting some check there...